src/app/api/api_v1/endpoints/welding_configuration.py

```diff
@@ -35,6 +37,38 @@ async def read_welding_configuration(
     return result
 
 
+@router.post("/upload", response_description="Upload file to create a new welding configuration",
+             response_model=WeldingConfiguration)
+async def upload_welding_configuration(
+        *,
+        db: AsyncSession = Depends(deps.get_async_db),
+        name: str,
+        file: UploadFile
+) -> Any:
+    parser = PandasParser(file)
+    if not parser.validate():
+        raise HTTPException(status_code=415, detail="File is not valid")
+
+    content = await file.read()
+    parse_result = parser.parse(content)
+    if parse_result.error:
+        raise HTTPException(status_code=415, detail=parse_result.detail)
+
+    # Create new welding configuration
+    welding_configuration_create = WeldingConfigurationCreate(name=name)
+    welding_configuration_obj = await welding_configuration.create(db=db, obj_in=welding_configuration_create)
+    if welding_configuration_obj is None:
+        raise HTTPException(status_code=500, detail="Could not create new welding configuration")
+
+    # Create welding points
+    for welding_point_obj in parser.get_welding_points(welding_configuration=welding_configuration_obj):
+        result = await welding_point.create(db=db, obj_in=welding_point_obj)
```

Comment:
I'd personally prefer some bulk creation at this point to avoid x Commits. Especially since it is undesirable for some of the welding points to get commited when there is an error.

Reply:
Addressed. Added support in crud/base for bulk creates. Also made a promising exploration with the `jsonable_encoder`.
And sorry: my pycharm wanted to reformat the whole base.py. I staged it for the commit as it looked valid, even though the commit is a bit uncleaner.